src/layouting/layouting.jl

```diff
@@ -21,16 +21,96 @@ function layout_text(
     pos = to_ndim(Point3f0, mpos, 0)
     scales = Vec2f0[glyph_scale!(atlas, c, ft_font, rscale) for c in string]
 
-    positions2d = calc_position(string, Point2f0(0), rscale, ft_font, atlas)
-    aoffset = align_offset(Point2f0(0), positions2d[end], atlas, rscale, ft_font, offset_vec)
-    aoffsetn = to_ndim(Point3f0, aoffset, 0f0)
-    positions = map(positions2d) do p
-        pn = rot * (to_ndim(Point3f0, p, 0f0) .+ aoffsetn)
-        pn .+ (pos)
+    glyphpos = glyph_positions(string, ft_font, rscale, offset_vec[1], offset_vec[2];
+        lineheight_factor = 1.5, justification = 0.0)
+
+    positions = Point3f0[]
+    for (i, group) in enumerate(glyphpos)
+        for gp in group
+            p = to_ndim(Point3f0, gp, 0) #./ Point3f0(4, 4, 1)
+            # rotate around the alignment point (this is now at [0, 0, 0])
+            p_rotated = rot * p
+            push!(positions, pos .+ p_rotated) # TODO why division by 4 necessary?
+        end
+        # between groups, push a random point for newline, it doesn't matter
+        # what it is
+        if i < length(glyphpos)
+            push!(positions, Point3f0(0, 0, 0))
+        end
     end
+
     return positions, scales
 end
 
+
+function glyph_positions(str::AbstractString, font, fontscale, halign, valign; lineheight_factor = 1.0, justification = 0.0)
+
+    # this is a countermeasure against Cairo messing with FreeType font pixel sizes
+    # when drawing. We reset them every time which is hacky but seems to work
+    FreeTypeAbstraction.FreeType.FT_Set_Pixel_Sizes(font, 64, 64)
```

Comment:
Shouldn't Cairo hacks go into CairoMakie?

Reply:
in principle yes but cairomakie affects freetype state.. so the countermeasure has to go there too. but maybe switching to a workflow without FT_Set_Pixel_Sizes works. It seems that what Cairo resets the fonts to is the standard size matching `font.units_per_EM`